subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_poll.c

```diff
@@ -383,9 +388,16 @@ static int start_download(void)
 		}
 	};
 
+	/* Clear semaphore before starting download */
+	k_sem_reset(&fota_download_sem);
+
+	fota_status = NRF_CLOUD_FOTA_IN_PROGRESS;
+	fota_status_details = NULL;
```

Comment:
This doesn't seem specific to this change, was it just missing before?

Draft LGTM so far

Reply:
you were only notified of the PR today, when it was taken out of draft.